paf-mvp-core-js/src/express/utils.ts

```diff
@@ -93,3 +105,31 @@ export const escapeRegExp = (stringForRegex: string): string => stringForRegex.r
  * @param host
  */
 export const getTopLevelDomain = (host: string) => domainParser(`https://${host}`).domain;
+
+export const extractRequestAndContextFromHttp = <
+  TopLevelRequestType,
+  TopLevelRequestRedirectType extends { returnUrl: ReturnUrl; request: TopLevelRequestType }
```

Comment:
FAR Those generic seems to be an accidental complexity. Maybe we can revamp at some point this part. WDYT?

For instance: 
```typescript
interface IdsAndPrefRequest {
    isRedirect: boolean,
    httpRequest: Request,
    redirectReturnUrl?: string,
    // ... 
}


export const extractContextFromHttp = (request: IdsAndPrefRequest): RestContext | RedirectContext => {
    if (request.isRedirect) {
        return {
            request,
            context: {
                returnUrl: request.returnUrl,
                referer: request.httpRequest.header('referer'),
            },
        };
    } else {
        return {
            request,
            context: { 
                origin: request.httpRequest.header('origin') 
            },
        };
    }
};
``` 

Or even simpler:
````typescript
interface IdsAndPrefRequest {
    isRedirect: boolean,
    httpRequest: Request,
    redirectReturnUrl?: string,
    context: RestContext | RedirectContext
    // ... 
}


class RedirectIdsAndPrefRequest implements IdsAndPrefRequest {
    public readonly isRedirect = true;
    public readonly httpRequest: Request;
    public readonly redirectReturnUrl: string;
    public get context(): RedirectContext {
        return {
            returnUrl: this.returnUrl,
            referer: this.httpRequest.header('referer'),
        }
    }

    constructor(httpRequest: Request, returnUrl: string) {
        this.httpRequest = httpRequest;
        this.redirectReturnUrl = returnUrl;
    }

}

class RestIdsAndPrefRequest implements IdsAndPrefRequest {
    public readonly isRedirect = false;
    public readonly httpRequest: Request;
    public get context(): RestContext {
        return {
            origin: this.httpRequest.header('origin')
        }
    }

    constructor(httpRequest: Request) {
        this.httpRequest = httpRequest;
    }

}
````

Reply:
I just moved the function to the utils file I didn't look at it in depth. 
It's true that it's not very readable. 
I am ok with sacrificing some of the genericity for the sake of readability. And I prefer your first suggestion. 
To be done in a later PR...